### PR TITLE
CIV-1916 extend response

### PR DESCRIPTION
**Pull request by TatsyMaslova**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-1916

### Change description ###
add new event for extending date 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — APPROVED** [review]
LGTM

**Review by laarmada — DISMISSED** [review]

**Review by neeta-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/InformAgreedExtensionDateForSpecCallbackHandler.java [hunk 1]
```diff
@@ -65,18 +68,22 @@ protected Map<String, Callback> callbacks() {
...
             callbackKey(MID, "extension-date"), this::validateExtensionDate,
+            callbackKey(MID, "response-deadline-extension"), this::validateProposedDeadlineAdmin,
             callbackKey(ABOUT_TO_SUBMIT), this::setResponseDeadline,
             callbackKey(SUBMITTED), this::buildConfirmation
         );
     }
 
     @Override
     public List<CaseEvent> handledEvents() {
+        if (toggleService.isPinInPostEnabled()) {
```
Comment (PaulH-HMCTS) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/InformAgreedExtensionDateForSpecCallbackHandler.java:79]: Look at whether we need to add any more unit/integration testing for this new functionality

Comment (TatsyMaslova) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/InformAgreedExtensionDateForSpecCallbackHandler.java:79]: Added test. Thank you Paul!

**Review by PaulH-HMCTS — DISMISSED** [review]

**Comment by TatsyMaslova** [comment]
reopen for testing

**Review by neeta-hmcts — APPROVED** [review]

**Review by nishm — APPROVED** [review]

**Pull request merged**